geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClear.java

```diff
@@ -396,6 +399,17 @@ void doClear(RegionEventImpl regionEvent, boolean cacheWrite) {
       // Force all primary buckets to be created before clear.
       assignAllPrimaryBuckets();
 
+      Iterator<AsyncEventQueue> allAEQsIterator =
+          partitionedRegion.getCache().getAsyncEventQueues(false).iterator();
+      while (allAEQsIterator.hasNext()) {
+        AsyncEventQueueImpl aeq = (AsyncEventQueueImpl) allAEQsIterator.next();
+        if (aeq.isPartitionedRegionClearSupported()) {
```

Comment:
the flag is confusing. As it is currently named "true" should mean that a PR region support clear. But the way you have it coded now "true" means we will throw an Unsupported exception. So do you want to change the property named to "PartitionedRegionClearUnsupported"?
Add some javadocs defining this property. If you set it to true then doing clear on a pr with that will throw the exception.
If you set to false then a clear on the pr will be allowed but will be ignored by the aeq/gateway.

Reply:
done